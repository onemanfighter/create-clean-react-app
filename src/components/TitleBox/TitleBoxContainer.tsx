import React from 'react';
import { Box } from '@chakra-ui/react';
import isEqual from 'lodash/isEqual';
import { TitleBoxProps } from './types';
import { Helmet } from 'react-helmet-async';

const TitleBoxContainer = ({
  title,
  icon,
  helmetChildren,
  ...props
}: TitleBoxProps) => {
  return (
    <Box {...props}>
      <Helmet>
        <title>{title}</title>
        {helmetChildren}
      </Helmet>
      {props.children}
    </Box>
  );
};

export default React.memo(TitleBoxContainer, isEqual);
