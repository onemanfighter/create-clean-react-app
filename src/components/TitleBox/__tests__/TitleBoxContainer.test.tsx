import { render } from '@testing-library/react';

import TitleBoxContainer from '../TitleBoxContainer';

const title = 'Test Title';
const icon = 'test-icon';
const props = {
  title,
  icon,
  children: 'Test Children',
};

describe('TitleBoxContainer', () => {
  it('should set the document title', () => {
    // Act
    const { container } = render(<TitleBoxContainer {...props} />);

    expect(container).toMatchSnapshot();
  });
});
