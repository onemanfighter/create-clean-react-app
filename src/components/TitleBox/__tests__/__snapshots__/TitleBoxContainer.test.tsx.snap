// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBoxContainer should set the document title 1`] = `
<div>
  <div
    class="css-0"
  >
    <div>
      <title>
        Test Title
      </title>
    </div>
    Test Children
  </div>
</div>
`;
